### ISSUE TITLE
time function for windows

### ISSUE THREAD

**Issue by fabiolimace** [f4b6a3/uuid-creator#53]

**Comment by fabiolimace** [comment]
Created `WindowsTimeFunction` to be used when the OS is WINDOWS.

On WINDOWS, the typical system time granularity is 15.625ms due to a default 64Hz timer frequency.

BENCHMARK:

```
Benchmark                 Mode  Cnt     Score    Error   Units
UuidCreator_TimeBased    thrpt    5  9999,914 ± 12,420  ops/ms
UuidCreator_TimeOrdered  thrpt    5  9999,544 ± 15,313  ops/ms
```

**Comment by fabiolimace** [comment]
Released version v4.2.0.
